libs/service/service.go

```diff
@@ -0,0 +1,25 @@
+package service
+
+import "sync/atomic"
+
+type State uint32
+
+const (
+	Stopped State = iota
+	Running
+)
+
+// Service provides an interface to change an check service state
+type Service struct {
+	state uint32
```

Comment:
?
```suggestion
	state State
```

Reply:
In this case storing and loading state will be more complex:
```
* atomic.StoreUint32((*uint32)(&s.state), uint32(state))
* return State(atomic.LoadUint32((*uint32)(&s.state)))
```

IMO, better to leave `uint32`

Reply:
Then maybe the naming of `state` here is inconsistent? `s.state` seems to be a key for the state, not a state? If so, it it's 1) not clearly named or documented, and 2) should be `uint64`, unless there's a very specific reason it must be a `uint32`.